ml-agents/mlagents/trainers/ghost/trainer.py

```diff
@@ -0,0 +1,134 @@
+    def should_update_elo_rating():
+        """
+        Should the elo rating be updated
+        :return: True if not ghost else False
+        """
+        return False
+
+    
+    def add_experiences(
+        self,
+        curr_all_info: AllBrainInfo,
+        next_all_info: AllBrainInfo,
+        take_action_outputs,
+    ):
+        """
+        No need to collect experiences since we don't update the ghost brain
+        """
+        pass
+
+    def process_experiences(self, current_info: AllBrainInfo, new_info: AllBrainInfo):
+        """
+        No need to collect experiences since we don't update the ghost brain
+        """
+        pass
+
+    def get_elo_rating(self):
+        return np.mean([policy.get_elo_rating() for policy in self.agent_policies.values()])
+
+    def get_action(self, curr_info: BrainInfo) -> ActionInfo:
+        """
+        Get an action using this trainer's current policy.
+        :param curr_info: Current BrainInfo.
+        :return: The ActionInfo given by the policy given the BrainInfo.
+        """
+        self.trainer_metrics.start_experience_collection_timer()
+        if len(curr_info.agents) == 0:
+            return self.policy.get_action(curr_info)
+
+        action_infos: List[ActionInfo] = [None] * len(curr_info.agents)
+        for idx, agent in enumerate(curr_info.agent_ids):
+            if agent not in self.agent_policies:
+                self.agent_policies[agent] = random.choice(self.policies)
+
+            agent_brain_info = curr_info.get_agent_brain_info(agent)
+            action_infos[idx] = self.agent_policies[agent].get_action(agent_brain_info)
+
+        result = combine(action_infos)
+        self.trainer_metrics.end_experience_collection_timer()
+        return result
+
+    def is_ready_update(self):
+        """
+        Ghost trainers are never updating
+        """
+        return False
+
+    def update_policy(self):
+        """
+        Ghost trainers are never updating
+        """
+        raise UnityTrainerException("The ghost trainer's policy shouldn't get updated.")
+
+    def save_model(self):
+        """
+        Don't save the model for ghost trainers
+        """
+        pass
+
+    def end_episode(self):
+        """
+        Get called when the academy resets.
+        Loads a random saved model for each policy
+        """
+        # Model path is the same as the master trainer's model path
+        ckpt_state = tf.train.get_checkpoint_state(self.policy.model_path)
+        if ckpt_state:
+            all_ckpts = tf.train.get_checkpoint_state(self.policy.model_path).all_model_checkpoint_paths
+
+            if all_ckpts:
+                for policy in self.policies:
+                    policy_used_ckpts = all_ckpts
+
+                    # There is a 1-p_load_from_all_checkpoints probability that we sample the policy only from the last load_from_last_N_checkpoints policies
+                    if random.random() > self.ghost_prob_sample_only_recent:
```

Comment:
@LeSphax Shouldn't this condition be reversed, according to the explanation? To me it seems that it'll have only 40% chances to pick a recent checkpoints list.

> ghost_prob_sample_only_recent: Probability that the ghost will only sample recent checkpoints instead of the whole history. 0.6 means 60% chance to sample recent and 40% to sample the whole history.

```suggestion
                    if random.random() < self.ghost_prob_sample_only_recent:
```